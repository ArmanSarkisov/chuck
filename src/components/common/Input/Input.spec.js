import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Input } from './index';

describe('Input test', () => {
  let component;
  let mockCallback;
  let mockObject;

  describe('Has props', () => {
    beforeEach(() => {
      mockCallback = jest.fn();
      mockObject = { mock: true };
      component = shallow(
        <Input
          id="id"
          register={mockCallback}
          control={mockObject}
          placeholder="placeholder"
          type="text"
        />,
      );
    });

    it('should render Card component', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render props type', () => {
      const input = component.find('input');
      expect(input.get(0).props.type).toBe('text');
    });

    it('should render props placeholder', () => {
      const input = component.find('input');
      expect(input.get(0).props.placeholder).toBe('placeholder');
    });

    it('should render props id', () => {
      const input = component.find('input');
      expect(input.get(0).props.id).toBe('id');
    });

    it('should render props placeholder', () => {
      const input = component.find('input');
      expect(input.get(0).props.control).toBe(mockObject);
    });

    it('should render props register', () => {
      expect(mockCallback.mock.calls.length).toBe(1);
    });
  });
});
