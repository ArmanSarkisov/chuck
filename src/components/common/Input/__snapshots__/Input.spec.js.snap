// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Has props should render Card component 1`] = `
<input
  className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
  control={
    Object {
      "mock": true,
    }
  }
  id="id"
  placeholder="placeholder"
  type="text"
/>
`;
